File: app/src/main/java/com/capstone/project_niyakneyak/data/patient_model/PatientData.java
package com.capstone.project_niyakneyak.data.patient_model;

import androidx.annotation.Nullable;
import androidx.lifecycle.LiveData;

import com.capstone.project_niyakneyak.data.alarm_model.Alarm;

import java.util.ArrayList;
import java.util.List;

public class PatientData {
    //Field
    private String patientName;
    private String patientAge;
    private List<MedsData> medsData;

    //Constructor
    public PatientData(){
        this.patientName = "Guest"; this.patientAge = null;
        medsData = new ArrayList<>();
    }
    public PatientData(@Nullable String patientName, @Nullable String patientAge){
        this.patientName = patientName; this.patientAge = patientAge;
        medsData = new ArrayList<>();
    }


    //Getter and Setter for patient Info.
    public void setPatientName(String patientName) {this.patientName = patientName;}
    public void setPatientAge(String patientAge) {this.patientAge = patientAge;}
    public String getPatientName() {return patientName;}
    public String getPatientAge() {return patientAge;}


    //Useful Functions for MedsData Configuration
    public void setMedsData(List<MedsData> medsData){this.medsData = medsData;}
    public List<MedsData> getMedsData(){return medsData;}
    public boolean searchMedsData(MedsData target){return medsData.contains(target);}
    public void addMedsData(MedsData data){this.medsData.add(data);}
    public boolean modifyMedsData(MedsData target, MedsData changed){
        if(!medsData.contains(target)) return false;

        MedsData data = medsData.get(medsData.indexOf(target));
        data.setID(changed.getID());data.setMeds_name(changed.getMeds_name());
        data.setMeds_detail(changed.getMeds_detail());
        data.setMeds_start_date(changed.getMeds_start_date());
        data.setMeds_end_date(changed.getMeds_end_date());
        data.setAlarms(changed.getAlarms());
        return true;
    }
    public boolean deleteMedsData(MedsData target){
        if(!medsData.contains(target)) return false;
        this.medsData.remove(target);
        return true;
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/data/patient_model/PatientData.kt
package com.capstone.project_niyakneyak.data.patient_model

class PatientData {
    //Getter and Setter for patient Info.
    //Field
    private var patientName: String?
    private var patientAge: String?
    private var medsData: MutableList<MedsData>

    //Constructor
    init {
        medsData = ArrayList()
    }
    constructor() {
        patientName = "Guest"
        patientAge = null
    }

    constructor(patientName: String?, patientAge: String?) {
        this.patientName = patientName
        this.patientAge = patientAge
    }

    constructor(patientName: String?, patientAge: String?, medsData: MutableList<MedsData>?) {
        this.patientName = patientName
        this.patientAge = patientAge
        if (medsData == null) this.medsData = ArrayList() else this.medsData = medsData
    }

    //Useful Functions for MedsData Configuration
    fun setMedsData(medsData: MutableList<MedsData>) {
        this.medsData = medsData
    }

    fun getMedsData(): List<MedsData> {
        return medsData
    }

    fun searchMedsData(target: MedsData): Boolean {
        return medsData.contains(target)
    }

    fun addMedsData(data: MedsData) {
        medsData.add(data)
    }

    fun modifyMedsData(target: MedsData, changed: MedsData): Boolean {
        if (!medsData.contains(target)) return false
        val data = medsData[medsData.indexOf(target)]
        data.medsName = changed.medsName
        data.medsDetail = changed.medsDetail
        data.medsStartDate = changed.medsStartDate
        data.medsEndDate = changed.medsEndDate
        data.alarms = changed.alarms
        return true
    }

    fun deleteMedsData(target: MedsData): Boolean {
        if (!medsData.contains(target)) return false
        medsData.remove(target)
        return true
    }
}